File: python/1876.substrings-of-size-three-with-distinct-characters.py
#
# @lc app=leetcode id=1876 lang=python
#
# [1876] Substrings of Size Three with Distinct Characters
#

# @lc code=start
class Solution(object):
    def countGoodSubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        count = 0
        n = len(s)

        for i in range(n - 2):
            if len(set(s[i:i+3])) == 3:
                count += 1
                
        return count


# @lc code=end

File: cpp/1876.substrings-of-size-three-with-distinct-characters.cpp
/*
 * @lc app=leetcode id=1876 lang=cpp
 *
 * [1876] Substrings of Size Three with Distinct Characters
 */

// @lc code=start
class Solution {
public:
    int countGoodSubstrings(string s) {

        // first check if the string is less than 3
        int n = s.size();
        if (n < 3) return 0;

        // var for counting the good substrings
        int count = 0;

        for (int i = 0; i < n - 2; i++) {
            if (s[i] != s[i + 1] && s[i + 1] != s[i + 2] && s[i] != s[i + 2]) {
                count++;
            }
        }

        return count;
    }
};
// @lc code=end

